item.go

```diff
@@ -0,0 +1,364 @@
+package automerge
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+	"unsafe"
+)
+
+// #include <automerge.h>
+import "C"
+
+// Kind represents the underlying type of a Value
+type Kind uint
+
+var (
+	// KindVoid indicates the value was not present
+	KindVoid Kind = C.AM_VAL_TYPE_VOID
+	// KindBool indicates a bool
+	KindBool Kind = C.AM_VAL_TYPE_BOOL
+	// KindBytes indicates a []byte
+	KindBytes Kind = C.AM_VAL_TYPE_BYTES
+	// KindCounter indicates an *automerge.Counter
+	KindCounter Kind = C.AM_VAL_TYPE_COUNTER
+	// KindFloat64 indicates a float64
+	KindFloat64 Kind = C.AM_VAL_TYPE_F64
+	// KindInt indicates an int
+	KindInt64 Kind = C.AM_VAL_TYPE_INT
+	// KindUint indicates a uint
+	KindUint64 Kind = C.AM_VAL_TYPE_UINT
+	// KindNull indicates an explicit null was present
+	KindNull Kind = C.AM_VAL_TYPE_NULL
+	// KindStr indicates a string
+	KindStr Kind = C.AM_VAL_TYPE_STR
+	// KindTime indicates a time.Time
+	KindTime Kind = C.AM_VAL_TYPE_TIMESTAMP
+	// KindUnknown indicates an unknown type from a future version of automerge
+	KindUnknown Kind = C.AM_VAL_TYPE_UNKNOWN
+
+	// KindMap indicates an *automerge.Map
+	KindMap Kind = kindObjType | C.AM_OBJ_TYPE_MAP
+	// KindList indicates an *automerge.List
+	KindList Kind = kindObjType | C.AM_OBJ_TYPE_LIST
+	// KindText indicates an *automerge.Text
+	KindText Kind = kindObjType | C.AM_OBJ_TYPE_TEXT
+
+	kindActorID     Kind = C.AM_VAL_TYPE_ACTOR_ID
+	kindDoc         Kind = C.AM_VAL_TYPE_DOC
+	kindChange      Kind = C.AM_VAL_TYPE_CHANGE
+	kindChangeHash  Kind = C.AM_VAL_TYPE_CHANGE_HASH
+	kindObjType     Kind = C.AM_VAL_TYPE_OBJ_TYPE
+	kindSyncState   Kind = C.AM_VAL_TYPE_SYNC_STATE
+	kindSyncMessage Kind = C.AM_VAL_TYPE_SYNC_MESSAGE
+)
+
+func (k Kind) String() string {
+	switch k {
```

Comment:
We can make a global unexported map for that 🤷

Reply:
:+1: